src/plugins/expressions/common/executor/executor.ts

```diff
@@ -255,21 +255,17 @@ export class Executor<Context extends Record<string, unknown> = Record<string, u
           if (transformedFn.type === 'function') {
             const prevChain = realIndex > 0 ? newAst.chain.slice(0, realIndex) : [];
             const nextChain = newAst.chain.slice(realIndex + 1);
-            return {
-              ...newAst,
-              chain: [...prevChain, transformedFn, ...nextChain],
-            };
+            newAst.chain = [...prevChain, transformedFn, ...nextChain];
```

Comment:
nit: why not using `splice` here?

Reply:
Contributions to the PR are more than welcome. I can't look at the particular logic of every loop in the 950 files I went through 🙂